### PR TITLE
Make `api` as a separate module

### LINKED ISSUE #745: Separate the API package as an independent module
In order to facilitate other projects referencing the [API package](https://github.com/streamnative/function-mesh/tree/master/api), it's better to separate the API package into an independent module. This approach can help reduce dependency conflicts.

### PR DESCRIPTION
**Pull request by jiangpengcheng**
*(If this PR fixes a github issue, please add `Fixes #<xyz>`.)*

Fixes #745 

*(or if this PR is one task of a github issue, please add `Master Issue: #<xyz>` to link to the master issue.)*

Master Issue: #<xyz>

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve.*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

Check the box below.

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [x] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

File: .github/workflows/bundle-release.yml [hunk 1]
```diff
@@ -56,7 +56,7 @@ jobs:
 
       - name: Install operator-sdk
         run: |
-          RELEASE_VERSION=v1.14.0
+          RELEASE_VERSION=v1.29.0
```
Comment (jiangpengcheng) [inline .github/workflows/bundle-release.yml:59]: this version will cause an error while run `make redhat-certificated-bundle`

```
Generating kustomize files in config/manifests
-: pattern /home/runner/work/function-mesh/function-mesh/api/...: main module (github.com/streamnative/function-mesh) does not contain package github.com/streamnative/function-mesh/api/compute/v1alpha1
FATA[0000] Error generating kustomize files: error getting ClusterServiceVersion base: error generating ClusterServiceVersion definitions metadata: one or more API packages had type errors
```

Comment (freeznet) [reply .github/workflows/bundle-release.yml:59]: any error from `make bundle`?

Comment (jiangpengcheng) [reply .github/workflows/bundle-release.yml:59]: it has same error if use v1.14.0, all caused by command:
```
operator-sdk generate kustomize manifests -q
```

Comment (freeznet) [reply .github/workflows/bundle-release.yml:59]: I cannot reproduce it, could you please delete the `bundle` directory and try again to see if any error occurs?

**Review by illegalnumbers — DISMISSED** [review]
I'm also confused as to why the `RELEASE_VERSION` of the operator has to get updated with this but the code change itself seems fine. cc @freeznet if you have a different opinion please mark a review here requesting changes.

**Comment by jiangpengcheng** [comment]
> I'm also confused as to why the `RELEASE_VERSION` of the operator has to get updated with this but the code change itself seems fine. cc @freeznet if you have a different opinion please mark a review here requesting changes.

the `RELEASE_VERSION` is not mean the "release version" for the function-mesh-operator, but the "release version" of the `operator-sdk`

File: .github/workflows/bundle-release.yml (hunk 1 above)
Comment (jiangpengcheng) [reply .github/workflows/bundle-release.yml:59]: yes, it's same error, did you run the command with this PR or on master branch?

**Review by freeznet — APPROVED** [review]

**Pull request merged**